engine.go

```diff
@@ -172,5 +172,5 @@ func (e *Engine) AddDatabase(db sql.Database) {
 
 // Init performs all the initialization requirements for the engine to work.
 func (e *Engine) Init() error {
-	return e.Catalog.LoadIndexes(e.Catalog.AllDatabases())
+	return nil // e.Catalog.LoadIndexes(e.Catalog.AllDatabases())
```

Comment:
No longer initializing indexes on engine init.  This is done per connection now.  Probably not optimal for performance.

Reply:
My main issue is with what this does to the interface. Clients have to have knowledge of the engine implementation details to initialize it properly. At the very least get rid of this method since it's not doing anything anymore.

Reply:
killed it.  It isn't even called anywhere in the go-mysql-server codebase, including the tests.